src/main/kotlin/de/bund/digitalservice/useid/statics/WidgetController.kt

```diff
@@ -1,15 +1,30 @@
 package de.bund.digitalservice.useid.statics
 
+import de.bund.digitalservice.useid.config.ContentSecurityPolicyProperties
+import org.springframework.http.server.reactive.ServerHttpRequest
+import org.springframework.http.server.reactive.ServerHttpResponse
 import org.springframework.stereotype.Controller
-import org.springframework.ui.Model
 import org.springframework.web.bind.annotation.GetMapping
 import reactor.core.publisher.Mono
 
 @Controller
-class WidgetController {
+class WidgetController(
+    private val contentSecurityPolicyProperties: ContentSecurityPolicyProperties
+) {
     @GetMapping("/widget")
-    fun widget(model: Model): Mono<String> = Mono.just("widget")
+    fun widget(serverHttpResponse: ServerHttpResponse, serverHttpRequest: ServerHttpRequest): Mono<String> {
+        val host = serverHttpRequest.uri.host
```

Comment:
This `host` is probably our domain, isn't it? Since it is the request URI. So this won't work. What we need is the domain of the origin of the request. So as described in [this ADR](https://github.com/digitalservicebund/useid-backend-service/blob/main/doc/adr/0005-csp-for-embedding-widget.md): "Therefore the widget.js file adds the domain of the parent page as query parameter to the iframe url."

Reply:
ah ok I missed that